homeassistant/components/zha/sensor.py

```diff
@@ -278,6 +301,53 @@ def native_unit_of_measurement(self) -> str:
         """Return Unit of measurement."""
         return self._channel.unit_of_measurement
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return device state attrs for battery sensors."""
+        attrs = {}
+        if self._channel.device_type is not None:
+            attrs["device_type"] = self._channel.device_type
+        status = self._channel.status
+        if status is not None:
+            attrs["status"] = str(status)[len(status.__class__.__name__) + 1 :]
+        return attrs
+
+
+@MULTI_MATCH(channel_names=CHANNEL_SMARTENERGY_METERING)
+class SmartEnergySummation(SmartEnergyMetering, id_suffix="summation_delivered"):
+    """Smart Energy Metering summation sensor."""
+
+    SENSOR_ATTR: int | str = "current_summ_delivered"
+    _device_class: str | None = DEVICE_CLASS_ENERGY
+    _state_class: str = STATE_CLASS_TOTAL_INCREASING
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return Unit of measurement."""
+        uom = self._channel.cluster.get("unit_of_measure")
+        if uom == 0:
+            return ENERGY_KILO_WATT_HOUR
+        return self._channel.unit_of_measurement
```

Comment:
This is correct for uom==0 when the metering_device_type==0. Falling back to the channel UOM for other values does not look right however. The UOMs defined for the channel all seem to relate to demand/rate. Summation measurements should generally not be per time unit. It would seem prudent to define corresponding summation UOMs for the summation attribute/sensor instead of falling back to demand/rate UOMs.

Reply:
Probably there can be a mapping.  kW->kWh etc.  If it's not one that HA has defined units for, then unitless is probably a better fallback than displaying something confusing.  ZCL defines unitless as 0x0B.

Reply:
ZCL also  states uom == 0 is kW and/or kWh. Users wanted W in the past for the instantaneous demand, cause kW was too coarse, that's fine. But for summation probable better to use kWh